document/CoreFormat.mdk

```diff
@@ -1445,6 +1453,36 @@ or if the tool cannot or chooses not to determine whether the result occurred at
 or if the tool examines source code written in a language that lacks such a construct, the `suppressionStates` array
 shall not include the value `"suppressedInSource"`.
 
+### `baselineState` property { #result-baselineState }
+
+If the `baselineId` property (&sect;[#run-baselineId]) of the current `run` object (&sect;[#run]) is present,
+then the `result` object may contain a property named `baselineState` whose value is a string that shall specify
+the state of this `result` with respect to the run specified by `baselineId`.
+
+If the `run.baselineId` property is absent, the `baselineState` property shall be absent.
+
+This property shall have one of the following values, with the specified meanings:
+
+* `"new"`: This result was detected in the current run but was not detected in the run specified by `run.baselineId`.
+
+* `"existing"`: This result was detected both in the current run and in the run specified by `run.baselineId`.
+
+* `"absent"`: This result was detected in the run specified by `run.baselineId` but was not detected in the current run.
+
+If the `run.baselineId` property is present but the `baselineState` property is absent, the `baselineState` property
+shall be considered to have the value `"new"`.
```

Comment:
IMO the presence of run.baselineId *means* that the results in this log have been compared to a previous log. Ah, but you're saying, "what if they want to compare to a log with no id"? I suggest we don't support that, for exactly the reason that I advocated for run.baselineId in the first place: If I have a log that says this issue is "new", but doesn't tell me "new since when", I don't think it's useful. As we discussed, in your ModernCop scenario, the user who runs the tool with the /baseline flag and examines the output immediately does know what the baseline was. But then if you store the log away and somebody looks at it later, they have no idea what the baseline was, unless run.baselineId is present.

---
In reply to: [63952132](https://github.com/sarif-standard/sarif-spec/pull/203#discussion_r63952132) [](ancestors = 63952132)